utils/s3/client.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
+
+public:
+    explicit client(ipv4_addr addr, private_tag);
```

Comment:
Will socket_address work?

Reply:
Should work, yes. Will fix

Reply:
Does seastar http code work with ipv6? If it works, s3::client should work with ipv6 as well, right?

Reply:
Seastar http code uses connected_socket whatever it is, so it would work with IPv6 too. I made this change in http, but apparently forgot to do the same with S3